Amber

njzjz opened the pull request.

njzjz committed:
AMBER

njzjz committed:
Merge branch 'devel' of github.com:deepmodeling/dpgen into amber

njzjz committed:
bugfix

njzjz committed:
do not use multiprocessing

njzjz committed:
retry slurm commands when failing
In my experience, sometimes dpgen fails to execute slurm commands. In this commit, dpgen will sleep 60 s and retry for at most 3 times.

njzjz committed:
retry slurm commands when failing
In my experience, sometimes dpgen fails to execute slurm commands. In this commit, dpgen will sleep 60 s and retry for at most 3 times.

njzjz committed:
Merge branch 'amber' of github.com:njzjz/dpgen into amber

njzjz committed:
fix bug

njzjz committed:
read parm7 from global jdata

njzjz committed:
read parm7 from global for fp

njzjz committed:
bugfix

Commit:
fix the bug of the units of taut and taup

amcadmus committed:
Merge pull request #296 from amcadmus/master

fix the bug of the units of taut and taup

njzjz committed:
bugfix

njzjz committed:
dump job record after every submitting

njzjz committed:
create dataset path before running

njzjz committed:
retry when ssh fails

njzjz committed:
Merge pull request #94 from deepmodeling/devel

merge devel

pep8speaks commented:
Hello @njzjz! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`dpgen/__init__.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/__init__.py):

> [Line 41:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/__init__.py#L41): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)

* In the file [`dpgen/dispatcher/Batch.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Batch.py):

> [Line 146:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Batch.py#L146): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)

* In the file [`dpgen/dispatcher/Dispatcher.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Dispatcher.py):

> [Line 232:72](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Dispatcher.py#L232): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 232:74](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Dispatcher.py#L232): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 232:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Dispatcher.py#L232): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 79 characters)
> [Line 232:114](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Dispatcher.py#L232): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','

* In the file [`dpgen/dispatcher/SSHContext.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py):

> [Line 72:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L72): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 261:5](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L261): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 261:42](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L261): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 261:44](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L261): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 263:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L263): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 265:67](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L265): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 268:70](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/SSHContext.py#L268): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`dpgen/dispatcher/Slurm.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Slurm.py):

> [Line 38:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Slurm.py#L38): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 199:5](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Slurm.py#L199): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 199:66](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Slurm.py#L199): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 204:44](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/dispatcher/Slurm.py#L204): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`dpgen/generator/run.py`](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py):

> [Line 320:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L320): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (130 > 79 characters)
> [Line 336:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L336): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 341:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L341): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 343:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L343): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 344:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L344): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 344:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L344): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 79 characters)
> [Line 345:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L345): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 347:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L347): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 348:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L348): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 348:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L348): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 79 characters)
> [Line 350:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L350): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 367:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L367): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (132 > 79 characters)
> [Line 470:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L470): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (128 > 79 characters)
> [Line 757:25](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L757): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 758:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L758): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 762:34](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L762): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 765:33](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L765): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 766:21](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L766): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 772:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L772): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (117 > 79 characters)
> [Line 772:81](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L772): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 772:83](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L772): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 772:97](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L772): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 772:99](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L772): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 792:50](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L792): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 793:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L793): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 802:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L802): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 804:44](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L804): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 806:42](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L806): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 808:49](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L808): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 810:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L810): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 812:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L812): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 828:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L828): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 829:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L829): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 836:58](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L836): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 836:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L836): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 837:25](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L837): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 837:63](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L837): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'
> [Line 840:36](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L840): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 841:21](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L841): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 844:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L844): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 79 characters)
> [Line 846:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L846): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 850:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L850): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 852:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L852): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 854:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L854): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 857:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L857): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 866:50](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L866): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 866:52](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L866): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 870:25](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L870): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 871:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L871): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 1125:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1125): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (158 > 79 characters)
> [Line 1131:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1131): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 1132:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1132): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 1133:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1133): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 1293:55](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1293): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1294:61](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1294): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1298:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1298): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 79 characters)
> [Line 1299:29](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1299): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 1299:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1299): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 1299:93](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1299): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1303:64](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1303): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'
> [Line 1303:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1303): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 1303:100](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1303): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'
> [Line 1307:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1307): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (98 > 79 characters)
> [Line 1307:96](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1307): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'
> [Line 1311:33](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1311): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1312:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1312): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (196 > 79 characters)
> [Line 1314:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1314): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 79 characters)
> [Line 1318:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1318): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 1323:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1323): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 1332:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1332): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 79 characters)
> [Line 1333:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1333): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 1355:36](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1355): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 1355:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1355): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 79 characters)
> [Line 1384:44](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1384): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ','
> [Line 1384:45](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1384): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 1385:11](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1385): [E271](https://duckduckgo.com/?q=pep8%20E271) multiple spaces after keyword
> [Line 1439:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1439): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 1922:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1922): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1927:40](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1927): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1938:56](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1938): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ','
> [Line 1938:57](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1938): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 1939:57](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1939): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ','
> [Line 1939:58](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1939): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 1941:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1941): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 1943:18](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1943): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 1944:25](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1944): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ','
> [Line 1944:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1944): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 1948:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1948): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 1951:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1951): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 1969:29](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L1969): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 2125:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2125): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 79 characters)
> [Line 2127:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2127): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 2128:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2128): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 2131:80](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2131): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 79 characters)
> [Line 2131:93](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2131): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 2131:95](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2131): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 2470:1](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2470): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 2478:26](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2478): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 2482:23](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2482): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 2491:27](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2491): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 2492:67](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2492): [E228](https://duckduckgo.com/?q=pep8%20E228) missing whitespace around modulo operator
> [Line 2494:16](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2494): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 2495:16](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2495): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 2495:44](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2495): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'
> [Line 2496:22](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2496): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 2496:48](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2496): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 2496:58](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2496): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 2498:20](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2498): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 2500:58](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2500): [E228](https://duckduckgo.com/?q=pep8%20E228) missing whitespace around modulo operator
> [Line 2502:54](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2502): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 2502:56](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2502): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 2521:29](https://github.com/njzjz/dpgen/blob/59d993cb9967b64b23abdfb9ac9ace47fbd420ab/dpgen/generator/run.py#L2521): [E203](https://duckduckgo.com/?q=pep8%20E203) whitespace before ':'



##### Comment last updated at 2021-06-30 04:18:47 UTC

njzjz committed:
support multiple RVAL

amcadmus committed:
Merge pull request #318 from deepmodeling/devel

merge devel into master

amcadmus committed:
Merge pull request #353 from deepmodeling/devel

merge recent development on devel to master

amcadmus committed:
Merge pull request #358 from deepmodeling/devel

Solve the comparability issue of the pymatgen v2022.0.*

Commit:
stop printing the dependencies, working around the issue of pymatgen.__version__

amcadmus committed:
Merge pull request #359 from amcadmus/master

stop printing the dependencies

njzjz committed:
Merge branch 'master' into amber

njzjz committed:
remove important_atom feature

njzjz committed:
refactor codes

njzjz committed:
link rst7 file

njzjz committed:
bugfix

njzjz committed:
Merge branch 'devel' into amber

njzjz committed:
update

njzjz committed:
clean codes

njzjz committed:
bugfix

njzjz committed:
bugfix

njzjz committed:
bugfix

njzjz committed:
bugfix

njzjz committed:
fp_candidate should be reversed

lol

njzjz committed:
check whether model_devi.out exists

njzjz committed:
fix "Argument list too long" for tar command

If the number of files is too much, it will raise an error
"Argument list too long". A solution is to split it to several
commands.

njzjz committed:
fix SSH session not active

njzjz committed:
Revert "retry when ssh fails"

This reverts commit cbc64c01ba62aaffa4eb0bbfe2d0215eaaa48397.

njzjz committed:
support task-wise restart for manual cuda devices

njzjz committed:
Merge branch 'devel' into amber

njzjz committed:
remove a directory asynchronously

njzjz committed:
bugfix

njzjz committed:
store sftp and reduce frequency of open_sftp

njzjz committed:
Merge branch 'devel' into amber

njzjz committed:
support deepmd 2.0

njzjz committed:
support dp 2.0

njzjz committed:
support dp 2.0

njzjz committed:
fix bugs of dispatcher

njzjz committed:
bugfix

njzjz committed:
slurm

njzjz committed:
update netcdf

njzjz commented on dpgen/generator/run.py:
```diff
@@ -317,13 +317,37 @@ def make_train (iter_index,
+            jinput['model']['fitting_net']['numb_aparam'] = 1
+            jinput['model']['fitting_net'].pop('numb_fparam', None)
+        else:
+            raise RuntimeError('invalid setting for use_ele_temp ' + str(use_ele_temp))
```
merged, pass

njzjz commented on dpgen/generator/run.py:
```diff
@@ -443,7 +467,7 @@ def run_train (iter_index,
         all_task.append(task_path)
     commands = []
-    if LooseVersion(mdata["deepmd_version"]) >= LooseVersion('1') and LooseVersion(mdata["deepmd_version"]) < LooseVersion('2'):
+    if LooseVersion(mdata["deepmd_version"]) >= LooseVersion('1') and LooseVersion(mdata["deepmd_version"]) < LooseVersion('3'):
```
all above merged pass

njzjz committed:
bugfix
